runtime/convertValues.go

```diff
@@ -551,6 +554,52 @@ func importDictionaryValue(
 	var dictionaryStaticType interpreter.DictionaryStaticType
 	if dictionaryType != nil {
 		dictionaryStaticType = interpreter.ConvertSemaDictionaryTypeToStaticDictionaryType(dictionaryType)
+	} else {
+		var keyStaticType, valueStaticType interpreter.StaticType
+
+		if len(keysAndValues) == 0 {
+			keyStaticType = interpreter.PrimitiveStaticTypeNever
+			valueStaticType = interpreter.PrimitiveStaticTypeNever
+		} else {
+			// Infer the keys type from the first entry
+
+			key := keysAndValues[0]
+			keyStaticType = key.StaticType()
+			keySemaType := inter.ConvertStaticToSemaType(keyStaticType)
+
+			if sema.IsSubType(keySemaType, sema.NumberType) {
+				keyStaticType = interpreter.PrimitiveStaticTypeNumber
+			} else if sema.IsSubType(keySemaType, sema.PathType) {
```

Comment:
What is the reason for specifically handling `Number` and `Path` subtypes here? Is this because `IsValidDictionaryKeyType` allows all subtypes of these types and we want to infer the supertype?

Reply:
Yes, exactly. Dictionary Keys could be a mix of numeric sub-types/ path sub-types, and can be still valid. So here we always go for the safest option, by inferring the most generic type.
i.e: sending a value similar to: `var map: {Number:String} = {1: "int", 2.3: "float"}`

Reply:
Makes sense 👍 

Could you please add this as a comment?

Reply:
Added in 6d7e59d